containerd-shim/process.go

```diff
@@ -186,23 +210,46 @@ func (p *process) create() error {
 		}
 		return err
 	}
-	p.stdio.stdout.Close()
-	p.stdio.stderr.Close()
+	if runtime.GOOS != "solaris" { // Temporary hack due to issues with pipes
+		p.stdio.stdout.Close()
+		p.stdio.stderr.Close()
+	}
 	if err := cmd.Wait(); err != nil {
 		if _, ok := err.(*exec.ExitError); ok {
 			return errRuntime
 		}
 		return err
 	}
-	data, err := ioutil.ReadFile("pid")
-	if err != nil {
-		return err
-	}
-	pid, err := strconv.Atoi(string(data))
-	if err != nil {
-		return err
+	if runtime.GOOS == "solaris" {
+		args := append([]string{
+			"--log", logPath,
+			"--log-format", "json",
+		}, p.state.RuntimeArgs...)
+		args = append(args, "start", p.id)
```

Comment:
I see, I guess given that Solaris has no need for an OOM Monitor this won't cause issue for now.